cpp/include/cudf/detail/gather.cuh

```diff
@@ -73,8 +73,8 @@ __global__ void gather_bitmask_kernel(table_device_view source_table,
         size_type destination_row = destination_row_base + threadIdx.x;
 
         const bool thread_active = destination_row < destination_col.size();
-        size_type source_row =
-          thread_active ? gather_map[destination_row] : 0;
+        size_type source_row = thread_active ?
+          static_cast<size_type>(gather_map[destination_row]) : 0;
```

Comment:
@shwina I wasn't sure if this is right -- isn't `gather_map` an `Iterator<T>` rather than an iterator of `size_type`?